src/components/recipe-form/RecipeForm.tsx

```diff
@@ -0,0 +1,207 @@
+"use client";
+
+import { useFieldArray, useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Plus, Trash2 } from "lucide-react";
+
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { Textarea } from "@/components/ui/textarea";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "@/components/ui/select";
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+
+import { recipeSchema, type Recipe } from "./schemas";
+
+export function RecipeForm() {
+  const form = useForm<Recipe>({
+    resolver: zodResolver(recipeSchema),
+    defaultValues: {
+      name: "",
+      category: "",
+      description: "",
+      ingredients: [{ name: "", quantity: "" }],
+      steps: [{ step: "" }],
```

Comment:
I think it would be nice to have:
Average price,
Difficulty level,
Time

Reply:
its not in db model so i didnt add it, we can extend it later